do/serverless.go

```diff
@@ -589,6 +680,93 @@ func (s *serverlessService) GetConnectedAPIHost() (string, error) {
 	return s.owClient.Config.Host, nil
 }
 
+// ReadProject takes the path where projec lies and validates the project.yml.
+// once project.yml is validated it reads the directory for all the files and sub-directory
+// and returns the struct of the files
+func (s *serverlessService) ReadProject(project *ServerlessProject, args []string) (ServerlessOutput, error) {
+	err := readTopLevel(project)
+	if err != nil {
+		return ServerlessOutput{}, err
+	}
+	_, err = readProjectConfig(project.ConfigPath)
+	if err != nil {
+		return ServerlessOutput{}, err
+	}
+	return ServerlessOutput{}, nil
```

Comment:
Even though this code path is hidden, since we're not doing anything with the `readProjectConfig` spec output here, it might make sense to return an error message like this until it's fully implemented (unless this is intended behavior):

```suggestion
	_, err = readProjectConfig(project.ConfigPath)
	if err != nil {
		return ServerlessOutput{}, err
	}
	return ServerlessOutput{}, fmt.Errorf("not implemented")
```